### ISSUE TITLE
[Switch] Issue- Screen readers announce "Button" instead of "Switch".

### ISSUE THREAD

**Issue by bauerbach** [radix-ui/themes#474]
I tested the Switch element with a screen reader and it read "Button" instead of "Switch". Also, the HTML is generating it as a button rather than a checkbox, and 'role="switch"' thus does not work.

**Comment by vladmoroz** [comment]
Switch implements the [Switch WAI-ARIA Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/switch/) to the letter.

It does use `role="switch"`, and the HTML element used should be mostly irrelevant when all semantics and interactions are handled. You can see switches based on a div, button, or input elements among the WAI-ARIA examples there.

If a particular screen reader reads it as "Button" rather than "Switch", it's a matter of how that screen reader implements this role. For the record, VoiceOver reads "switch, on" and "switch, off" correctly.
